core/src/main/java/org/jdbi/v3/core/statement/SqlStatements.java

```diff
@@ -26,6 +27,10 @@
     private StatementRewriter statementRewriter;
     private TimingCollector timingCollector;
 
+    private boolean returningGeneratedKeys;
+    private String[] generatedKeysColumnNames;
```

Comment:
I wonder if it makes sense to declare the generated columns as a `List` rather than an array. We could then force its immutability via `unmodifiable` decorators rather than copying the array.